service_manager/example/lib/plugin_example/service_manager_page.dart

```diff
@@ -13,15 +13,32 @@ class _ServiceManagerWidgetState extends State<ServiceManagerWidget> {
   @override
   void initState() {
     super.initState();
-    _isBluetoothEnabled();
+    _initBluetooth();
   }
 
-  Future<void> _isBluetoothEnabled() async {
+  Future<void> _initBluetooth() async {
     bool bluetoothEnabled = await isBluetoothEnabled();
 
     setState(() {
       _bluetoothEnabled = bluetoothEnabled;
     });
+
+    state.listen((state) {
+      switch (state) {
+        case 10:
+          setState(() {
+            _bluetoothEnabled = false;
+          });
+          break;
+        case 12:
+          setState(() {
+            _bluetoothEnabled = true;
+          });
+          break;
```

Comment:
These are quite magic mysterious numbers. I would create an enum for the different states. These numbers are from Android, but these are probably different for iOS.